pkg/kgo/compression.go

```diff
@@ -233,9 +278,13 @@ type decompressor struct {
 	ungzPool   sync.Pool
 	unlz4Pool  sync.Pool
 	unzstdPool sync.Pool
+	pools      pools
 }
 
-func newDecompressor() *decompressor {
+// DefaultDecompressor returns the default decompressor used by clients.
+// If a pool is provided and implements PoolDecompressBytes, it will be
+// used.
+func DefaultDecompressor(pools ...Pool) Decompressor {
```

Comment:
is the intention here that only one pool is passed (and the function takes a variadic because that pool is optional)? It could be just a language thing, but that wasn't immediately clear to me.

Reply:
The first pool that implements PoolDecompressBytes will be used; the intention here is to allow the end user to use the same pools here that they use in the config option `WithPools` -- so, they can be a bit lazy if they have multiple separate pools rather than figure out which pool they should use.

Reply:
ahaaa. Is it the first or the last? Currently the loop in `decompressor.Decompress` takes a slice for all implementations and keeps only the last one. 

Maybe mention this behaviour in a brief comment here as this is a public function (or a validation function and not rely on humans/AI reading comments),

Reply:
Actually you're pointing out a bug -- the first pool that returns a slice should cause the `each` function to return early. I'm going to change `each` arg to `func(Pool) bool`, and if it returns true, stop iterating.